### ISSUE TITLE
make region argument optional in hydromt build 

### ISSUE THREAD

**Issue by DirkEilander** [Deltares/hydromt#269]
Currently the region argument is required both in the `hydromt build` CLI and the `Model.build()` method. This might nog always be wanted, for instance when initializing a grid model with its grid properties as below (this is a use case of the SfincsModel). Also it prohibits from setting the region in the hydromt config file which might for some applications also be more straightforward especially when used in combination with other hydrography data.

_Proposed implementation_
- We provide region via CLI we use a keyword `--region` (NOTE: `-r` is already in use for resolution). This means that an error message will be thrown if a region is passed without the keyword, but I think everyone will adapt quickly. It's not possible as far as I can see to support region both as argument and optional keyword at the same time.
- The build we make region optional and we only add this argument to `opt` if provided.

_Example setup_grid without region argument_
```
[setup_grid]
width = 494
height = 373
dx = 50.0
dy = 50.0
x0 = 264450.0
y0 = 5073200.0
rotation = 0
crs = 32633
```

**Comment by B-Dalmijn** [comment]
TODO:

- [ ] remove res from cli build, make region optional 
- [ ] idem in model_api build functions
- [ ] add nice warning message if old -r is used & add deprecation in help

**Comment by DirkEilander** [comment]
this is fixed in #278
